### PR TITLE
Enabled refit on Python 3.13

### PR DESCRIPTION
**Pull request by cehongwang**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/_features.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 _TORCHTRT_RT_AVAIL = _TS_FE_AVAIL or os.path.isfile(linked_file_runtime_full_path)
 _DYNAMO_FE_AVAIL = version.parse(sanitized_torch_version()) >= version.parse("2.1.dev")
 _FX_FE_AVAIL = True
-_REFIT_AVAIL = version.parse(sys.version.split()[0]) < version.parse("3.13")
+_REFIT_AVAIL = version.parse(sys.version.split()[0]) < version.parse("3.14")
```
Comment (narendasan) [inline py/torch_tensorrt/_features.py:40]: Just say True if it works

File: py/torch_tensorrt/dynamo/_refit.py [hunk 2]
```diff
@@ -118,13 +89,12 @@ def construct_refit_mapping_from_weight_name_map(
...
     engine_weight_map = {}
     for engine_weight_name, (sd_weight_name, np_weight_type) in weight_name_map.items():
-        trt_dtype = dtype.try_from(np_weight_type).to(trt.DataType)
-        torch_dtype = dtype.try_from(np_weight_type).to(torch.dtype)
-
         if sd_weight_name not in state_dict:
             # If weights is not in sd, we can leave it unchanged
             continue
         else:
+            trt_dtype = dtype.try_from(np_weight_type).to(trt.DataType)
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/_refit.py:96]: try_from needs handling since it may return None. If you want the exception use from

File: py/torch_tensorrt/dynamo/_refit.py [hunk 3]
```diff
@@ -208,8 +178,9 @@ def _refit_single_trt_engine_with_gm(
                 if layer_name not in mapping:
                     raise AssertionError(f"{layer_name} is not found in weight mapping")
                 # Use Numpy to create weights
-                weight, datatype = mapping[layer_name]
-                trt_wt_tensor = trt.Weights(datatype, weight.ctypes.data, weight.size)
+                weight = mapping[layer_name]
+                trt_dtype = dtype.try_from(weight.dtype).to(trt.DataType)
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/_refit.py:182]: Same here

File: py/torch_tensorrt/dynamo/_refit.py [hunk 4]
```diff
@@ -53,7 +52,7 @@ def construct_refit_mapping(
     module: torch.fx.GraphModule,
     inputs: Sequence[Input],
     settings: CompilationSettings = CompilationSettings(),
-) -> dict[str, np.ndarray]:
+) -> Any:
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/_refit.py:55]: Can we not be a little more specific?
